usecases/replica/coordinator.go

```diff
@@ -190,38 +213,86 @@ func (c *coordinator[T]) Pull(ctx context.Context,
 	}
 	level := state.Level
 	replyCh := make(chan _Result[T], level)
-
-	candidates := state.Hosts[:level]                          // direct ones
-	candidatePool := make(chan string, len(state.Hosts)-level) // remaining ones
-	for _, replica := range state.Hosts[level:] {
-		candidatePool <- replica
-	}
-	close(candidatePool) // pool is ready
+	hosts := state.Hosts
 	f := func() {
+		hostRetryQueue := make(chan hostRetry, len(hosts))
+
+		// put the "backups/fallbacks" on the retry queue
+		for i := level; i < len(hosts); i++ {
+			hostRetryQueue <- hostRetry{
+				hosts[i],
+				backoff.WithContext(utils.NewExponentialBackoff(c.pullBackOffPreInitialInterval, c.pullBackOffMaxElapsedTime), ctx),
+			}
+		}
```

Comment:
**question:** is this needed ? given that we push the retry in https://github.com/weaviate/weaviate/pull/5472/files#diff-8fbb63407bec007b00f3be803b48b8d5bea97f07fc349722ed4e7fde2cf3e0c9R249 ?

Reply:
Good question, the code isn't as clean as I'd like, but these two retry pushes are both needed in this because the first push handles only the single op try for each worker, while the second push handles failures on retries. It would be nice to unify these into a single loop, but ensuring the fullRead could change hosts on each attempt made the code to handle this more complex than just duplicating the op call/error handling. I'll look into improving the code cleanliness in a follow up PR though